metals/src/main/scala/scala/meta/internal/metals/MetalsEnrichments.scala

```diff
@@ -381,8 +381,9 @@ object MetalsEnrichments
     }
 
     def isDependencySource(workspace: AbsolutePath): Boolean =
-      (isLocalFileSystem(workspace) &&
-        isInReadonlyDirectory(workspace)) || isJarFileSystem
+      isLocalFileSystem(workspace) &&
+        (isInReadonlyDirectory(workspace) || isInTmpDirectory(workspace)) ||
```

Comment:
`.metals/.tmp/Main-...` aren't actually dependency sources, right? Maybe we could add a new method `isDependencySourceOrTmp`?